src/emulate.c

```diff
@@ -395,36 +398,47 @@ enum {
 #undef _
 };
 
+/* FIXME: This will simply find the n-th instruction by iterating
+ * the linked list linearly, we may want to find better approach. */
+#define GET_NEXT_N_INSN(ir, n)                   \
+    ({                                           \
```

Comment:
`({` and `})` denotes the the statement expression, which is a feature of GCC/Clang. See https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html
Please avoid it by rewriting into an inline function.